### PR TITLE
koji-tools types

### PR DESCRIPTION
**Pull request by bdjeffyp**
If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/koji-tools/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export as namespace Koji;
+
+declare namespace Koji {
+    // index.js methods
+    /**
+     * Sets up Koji.config parameters for each client and handles communication
+     * between the Koji live preview iframe and your app.
+     * @param options ???
+     */
+    function pageLoad(options?: {}): void;
```
Comment (sheetalkamat) [inline types/koji-tools/index.d.ts]: Missing few properties eg. from [usage](https://github.com/madewithkoji/koji-tools#usage---frontend) `Koji.config.colors.backgroundColor` and `Koji.config.strings.content` is suppose to exist?

File: types/koji-tools/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    function getConfig(): any;
+
+    /**
+     * Registers a callback on a Koji event.
+     * @param event name of the event being called
+     * @param callback method to execute when the event occurs
+     */
+    function on(event: string, callback: () => void): void;
+
+    function callEvent(event: string, params?: []): void;
```
Comment (sheetalkamat) [inline types/koji-tools/index.d.ts]: Missing more properties as per https://github.com/madewithkoji/koji-tools#api

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/koji-tools/index.d.ts (hunk 1 above)
Comment (bdjeffyp) [reply types/koji-tools/index.d.ts]: The `Koji.config` object is created from various JSON files into one JSON string, as specified in `./tools/buildConfig.js`. This makes any subelements of `Koji.config` and their types unknown. In the example you cited, the config object was built with two JSON files, named `colors.json` and `strings.json`. `buildConfig.js` bundles those JSON objects into the `Koji.config` object.

Comment (bdjeffyp) [reply types/koji-tools/index.d.ts]: This is why the `getConfig()` method returns an `any` type.

Comment (bdjeffyp) [reply types/koji-tools/index.d.ts]: I have changed the definition file to use `config` and specified it as an `object` type.

File: types/koji-tools/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+// Type definitions for koji-tools 0.5
+// Project: https://github.com/madewithkoji/koji-tools
+// Definitions by: Jeff Peterson <https://github.com/bdjeffyp>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+// tslint:disable-next-line:export-just-namespace
+export = Koji;
```
Comment (sheetalkamat) [inline types/koji-tools/index.d.ts]: Do not disable this. Remove namespace declaration and export the declarations inside it directly.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/koji-tools/index.d.ts (hunk 3 above)
Comment (bdjeffyp) [reply types/koji-tools/index.d.ts]: Done!

**Pull request merged**